src/components/ProductListing/ProductListing.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import { useEffect } from "react";
+
+import { useProduct } from '../../context/productcontext';
+import axios from 'axios';
+
+import "./ProductListing.css"
+export const ProductListing = () => {
+    const { product,
+        setProduct, sortBy, category, rating, range } = useProduct();
+
+    useEffect(() => {
```

Comment:
suggestion - Can use try and catch it will help in handle the error in fetch data